node/bindnode/repr.go

```diff
@@ -1026,7 +1026,8 @@ func (w *_structIteratorRepr) Next() (key, value datamodel.Node, _ error) {
 		keyStr, _ := key.AsString()
 		mappedKey := outboundMappedKey(stg, keyStr)
 		if mappedKey != keyStr {
-			key = basicnode.NewString(mappedKey)
+			w.keyStringAddr = mappedKey
+			key = basicnode.NewStringAddr(&w.keyStringAddr)
```

Comment:
Similarly, a short comment on this iterator (or somewhere that's discoverable?) about how it's *not safe* to use this node elsewhere and assume that it's immutable seems very important.  The norm is to assume that a node is immutable and you can stitch it together in larger COW structures without second thoughts.